dds_web/web/user.py

```diff
@@ -147,35 +138,103 @@ def register():
         except (sqlalchemy.exc.SQLAlchemyError, sqlalchemy.exc.IntegrityError) as sqlerr:
             raise ddserr.DatabaseError from sqlerr
 
-        # Go to two factor authentication setup
-        # TODO: Change this after email is introduced
-        flask_login.login_user(new_user)
-        return flask.redirect(flask.url_for("auth_blueprint.two_factor_setup"))
-
     # Go to registration form
     return flask.render_template("user/register.html", form=form)
 
 
-@auth_blueprint.route("/login", methods=["GET", "POST"])
+@auth_blueprint.route("/cancel_2fa", methods=["POST"])
 @limiter.limit(
     dds_web.utils.rate_limit_from_config,
     methods=["POST"],
     error_message=ddserr.error_codes["TooManyRequestsError"]["message"],
 )
-def login():
-    """Log user in with DDS credentials."""
+def cancel_2fa():
+    form = forms.Cancel2FAForm()
+    # Only checks the csrf I assume (doesn't hurt to validate)
+    if form.validate_on_submit():
+        # Reset 2fa cookie
+        redirect_to_login = flask.redirect(flask.url_for("auth_blueprint.login"))
+        redirect_to_login.set_cookie("2fa_initiated_token", "", expires=0)
+        return redirect_to_login
+
+    return flask.redirect(flask.url_for("auth_blueprint.login"))
```

Comment:
lines 154-160 look a little bit strange. It seems like you could rewrite it even as 

```
form.validate_on_submit()
redirect_to_login = flask.redirect(flask.url_for("auth_blueprint.login"))
        redirect_to_login.set_cookie("2fa_initiated_token", "", expires=0)
        return redirect_to_login
```

because no matter the outcome of validation you redirect to the same endpoint and might as well unset the cookie in any case?
And it seems like validate only returns a boolean, not that it raises an exception. Does it help us in any way?

Reply:
I opted for Inas solution.